packages/metrics/src/Metrics.ts

```diff
@@ -381,8 +383,11 @@ class Metrics extends Utility implements MetricsInterface {
         service: this.defaultDimensions.service,
       });
     }
-    if (this.functionName != null) {
-      singleMetric.addDimension('function_name', this.functionName);
+    for (const value of [this.functionName, functionName]) {
+      if (value && value.trim().length > 0) {
+        singleMetric.addDimension('function_name', value);
+        break;
+      }
```

Comment:
```suggestion
	const value = this.functionName?.trim() || functionName?.trim();
	if (value && value.length > 0) {
	  singleMetric.addDimension('function_name', value);
	}
```

Isn't this easier to read and understand?